packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -0,0 +1,539 @@
+    if (this.changeOutputIndex !== undefined) {
+      return this.outputs[this.changeOutputIndex];
+    }
+    return null;
+  }
+
+  // Protected methods
+
+  /**
+   * Analyzes the inputs of the transaction.
+   * @returns {TransactionInput[]} An array of analyzed transaction inputs
+   * @protected
+   */
+  protected analyzeInputs(): TransactionInput[] {
+    return this.originalTx.ins.map((input) => ({
+      txid: input.hash.reverse().toString("hex"),
+      vout: input.index,
+      sequence: input.sequence,
+      scriptSig: input.script.toString("hex"),
+      witness: input.witness.map((w) => w.toString("hex")),
+    }));
+  }
+
+  /**
+   * Analyzes the outputs of the transaction.
+   * @returns {TransactionOutput[]} An array of analyzed transaction outputs
+   * @protected
+   */
+  protected analyzeOutputs(): TransactionOutput[] {
+    return this.originalTx.outs.map((output, index) => ({
+      value: output.value,
+      scriptPubKey: output.script.toString("hex"),
+      address: this.getOutputAddress(output.script),
+      isChange: index === this.changeOutputIndex,
+    }));
+  }
+
+  /**
+   * Attempts to derive the address from an output script.
+   * @param {Buffer} script - The output script
+   * @returns {string} The derived address or an error message if unable to derive
+   * @protected
+   */
+  protected getOutputAddress(script: Buffer): string {
+    try {
+      const p2pkhAddress = payments.p2pkh({
+        output: script,
+        network: networkData(this.network),
+      }).address;
+      const p2shAddress = payments.p2sh({
+        output: script,
+        network: networkData(this.network),
+      }).address;
+      const p2wpkhAddress = payments.p2wpkh({
+        output: script,
+        network: networkData(this.network),
+      }).address;
+      const p2wshAddress = payments.p2wsh({
+        output: script,
+        network: networkData(this.network),
+      }).address;
+
+      return (
+        p2pkhAddress ||
+        p2shAddress ||
+        p2wpkhAddress ||
+        p2wshAddress ||
+        "Unable to derive address"
+      );
+    } catch (e) {
+      return "Unable to derive address";
+    }
+  }
+
+  /**
+   * Calculates the fee for the transaction.
+   * @returns {BigNumber} The calculated fee in satoshis
+   * @protected
+   */
+  protected calculateFee(): BigNumber {
+    return new BigNumber(this.currentFeeRate).multipliedBy(this.vsize);
+  }
+
+  /**
+   * Checks if the transaction signals RBF (Replace-By-Fee).
+   * @returns {boolean} True if the transaction signals RBF, false otherwise
+   * @protected
+   */
+  protected isRBFSignaled(): boolean {
+    return this.originalTx.ins.some((input) => input.sequence < 0xfffffffe);
+  }
+
+  /**
+   * Determines if RBF (Replace-By-Fee) can be performed on this transaction.
+   * @returns {boolean} True if RBF can be performed, false otherwise
+   * @protected
+   */
+  protected canPerformRBF(): boolean {
+    return this.isRBFSignaled();
```

Comment:
OR if the user/consumer decides they don't care and want to assume full rbf

Reply:
added new internal property of `_assumeRBF` which can be toggled by the user code using a setter